From ea0cae0ff7feb42f8a1cb5d6b9211f84acd94cca Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Fri, 28 Mar 2025 23:57:59 +0100
Subject: [PATCH] Avoid network access to get Electron runtime checksums

Avoid hitting the network to verify the runtime downloads by providing
them ahead of time. Should fix a build failure on newer versions of
Electron.

See: https://github.com/flathub/tech.feliciano.pocket-casts/pull/11#issuecomment-2758536925
---
 package.json | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/package.json b/package.json
index a4531457cad9..de0bab88eceb 100644
--- a/package.json
+++ b/package.json
@@ -68,7 +68,14 @@
 	},
 	"config": {
 		"forge": {
-			"packagerConfig": {},
+			"packagerConfig": {
+				"download": {
+					"checksums": {
+						"electron-v33.3.1-linux-x64.zip": "cdca6f9ec2f98708dcbcb8baba3d0521e15ed4e5aab68b2d1c6668f99faafc38",
+						"electron-v33.3.1-linux-arm64.zip": "b428ecca13d4ffc05c28e28759a310f317e4e89b05e9a30ba07e0d58fbaedef5"
+					}
+				}
+			},
 			"makers": [
 				{
 					"name": "@electron-forge/maker-zip",
-- 
2.49.0

